Integration of ICT Electricity Demand as Separate End-Use

hagento opened the pull request:
This PR integrates exogenous ICT (Information and Communication Technology) electricity demand projections as a distinct end-use category in the EDGE-Buildings model, separating it from the appliances & lighting category where it was previously included.

In `integrateICT()` the following processing steps happen:

 - Historical ICT shares relative to total appliances & lighting (A&L) electricity demand are calculated using the exogenous ICT data from `mredgebuildings::calcElecDemandICT` (see [PR](https://github.com/pik-piam/mredgebuildings/pull/79))
 - Missing historical periods  are filled using constant ICT/A&L shares
 - Future ICT projections are directly incorporated from the exogenous scenario data according to the selected techScen parameter (e.g. `SSP2_lower`)
 - The separated ICT demand is subtracted from A&L electricity to avoid double-counting

This results in the following changes to the electricity demand of `appliances_light`:

<img width="3000" height="2250" alt="EDGEB_AppliancesComparison" src="https://github.com/user-attachments/assets/6fac23ae-ca70-4945-900f-ea5420580acf" />

The `itc` projections compare to `appliances_light` as:

<img width="3000" height="2250" alt="EDGEB_SSPs" src="https://github.com/user-attachments/assets/e64d504e-a0f4-4bf8-9905-a93d96008a83" />


What is directly noticeable is that the demand hierarchy with respect to SSPs is contrary to ours, which we might want to discuss further.

Commit:
integrate ict elec demand and full PRISMA scenarios

Commit:
change ICT demand extrapolation to constant 2050 level

hagento commented:
I just committed some changes that change the extrapolation beyond 2050 to a constant ICT demand rather than any linear extrapolation etc since this seems to be the way we are going now.

robinhasse commented on inst/config/configPRISMA_ICT.csv:
Please add a blank terminal line.

robinhasse commented on inst/config/default.csv:
Please add a blank terminal line.

robinhasse commented on R/integrateICT.R:
```diff
@@ -0,0 +1,137 @@
+
+  } else {
+
+    ## Extrapolate future ICT demand with constant 2050 demand ====
```
I was irritated to not find 2050 explicitly in the code. Tiny change to make this a bit clearer perhaps.
```suggestion
    ## Extrapolate future ICT demand constantly ====

    # original projections run only until 2050 
```

robinhasse reviewed (changes requested):
Thanks Hagen, I like it. Just left some rather technical suggestions no substantial objections.

hagento commented on inst/config/default.csv:
Sorry about this again - I think I have finally found a csv Editor that prevents this from happening.

hagento commented on R/integrateICT.R:
```diff
@@ -0,0 +1,137 @@
+      left_join(ictComplete %>% select(.data$region, .data$period, .data$value),
+                by = c("region", "period"),
+                suffix = c("", "ICT")) %>%
+      mutate(value = .data$value - .data$valueICT, .keep = "unused")
```
I gave some thought to this as well - generally it is, of course, possible since ICT demands are given exogenously and we could potentially stumble upon a case where `ratio > 1`. On our current data, this is luckily not happening. Maybe for now we just include a warning and hope this does not pop up after future input data changes? 😬

hagento commented on R/integrateICT.R:
```diff
@@ -0,0 +1,137 @@
+
+    fe <- fe %>%
+      filter(.data$variable != "appliances_light.elec|fe") %>%
+      rbind(extrapolatedDemand)
```
Looking at the code now, I don't know what I was thinking when I implemented this mess - let's stick with `interpolate_missing_periods`.

hagento committed:
few code style changes

Co-authored-by: Robin Hasse <76682203+robinhasse@users.noreply.github.com>

Commit:
assume FE and UE equality for ICT

robinhasse reviewed (approved):
Thanks Hagen!

Commit:
added convergence of ICT demand towards 2050 target and extrapolation until 2100

Commit:
correct output variables for ICT

Commit:
include ICT in visualizeScenarios

Commit:
updated configPRISMA_ICT and added SSP5

Commit:
increment package version

hagento commented:
@robinhasse Considering PR #48, I would most likely remove `config_prismaICT.csv` and set the default in `config_remind.csv` to `techScen = <SSP>_medium`. But I'd like to do this after the changed here are approved since I'm also still actively working with the PRISMA config.

robinhasse commented:
> This formulation pulls down high per-capita regions more strongly than it pulls up low per-capita regions, while preserving the population-weighted global average.

How is the implementation preserving the global average? Does this result from the specific parameters? if so, I'd rather rescale to make sure this also works if numbers change.

robinhasse reviewed (approved):
Thanks, looks good to me. I'm okay with having the numbers hard-written into the code. Though I'm not as sure that we won't touch them again soon. I have just one question regarding the regional convergence.

robertpietzcker commented:
> > This formulation pulls down high per-capita regions more strongly than it pulls up low per-capita regions, while preserving the population-weighted global average.
> 
> How is the implementation preserving the global average? Does this result from the specific parameters? if so, I'd rather rescale to make sure this also works if numbers change.

I am not sure that is worth the effort at the moment, @robinhasse 
The point was that we got some projections in PRISMA that I deemed problematic in their regional split (factor 200 difference between regions in per-capita demand sustained until 2050), so we proposed a regional convergence but needed to make sure that the overall number stayed roughly the same to be compliant with this one project. 

if we come up with a different input, then we can rethink if we want to keep the convergence/ keep the global value unchanged, and make that change by hand. 
Coding a fail-safe algorithm that produces reasonable convergence while maintaining global values would be more effort, I would say - especially when we don't know WHAT type of new projections we will get in the future.

Commit:
merge configPRISMA_ICT into config_remind

Commit:
rescale to 2050 global ict demand

hagento commented:
I added one short line to make sure that we preserve the 2050 global demanded the impacts were minimal so I added the code to the PR and would merge everything. Maybe later on we can think of some more thorough convergence methods. 
<img width="1524" height="763" alt="Bildschirmfoto 2026-04-21 um 17 11 09" src="https://github.com/user-attachments/assets/166565c6-b424-4e78-8230-56b7ec1f2435" />
